homeassistant/components/media_player/denonavr.py

```diff
@@ -49,6 +53,7 @@
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
     vol.Optional(CONF_HOST): cv.string,
     vol.Optional(CONF_NAME): cv.string,
+    vol.Optional(CONF_SOUND_MODE, default=DEFAULT_SOUND_MODE): cv.boolean,
```

Comment:
Why is sound mode configurable? Can it not be detected from the device?

Reply:
This is the option to disable sound_mode_support for old types receivers that do not have sound modes. As far as I know this can not be detected from the receiver itself. At least their is no direct bolean if sound mode is supported. It might be possible to see it from the absense of the current sound mode if you request it from the receiver.

However I always like to have the option to disable components in case they give problems for certain types/models. Of course I only have my own receiver to test with and that one does support sound_mode. So I cann't really test with a receiver that doesn't support sound mode.

Reply:
> However I always like to have the option to disable components in case they give problems for certain types/models. 

In Home Assistant we aim to represent devices as-is. If we go down the route to disable sound modes, we can start adding config options for everything…

Please remove the config option and base it on detecting support.

Reply:
Okay. This PR can be merged when the config option has been removed and the lib has been updated.